utils/build.ps1

```diff
@@ -2405,7 +2405,7 @@ function Build-FoundationMacros() {
     -UseBuiltCompilers Swift `
     -SwiftSDK (Get-SwiftSDK $Platform) `
     -Defines @{
-      SwiftSyntax_DIR = (Get-ProjectCMakeModules $HostArch Compilers);
+      SwiftSyntax_DIR = (Get-ProjectCMakeModules $Arch Compilers);
```

Comment:
I think that we should assert `($Arch -eq $HostArch) -or ($Arch -eq $BuildArch)`.